api/pkg/apis/v1alpha1/managers/campaigns/campaigns-manager.go

```diff
@@ -126,14 +160,28 @@ func (m *CampaignsManager) DeleteState(ctx context.Context, name string, namespa
 	var err error = nil
 	defer observ_utils.CloseSpanWithError(span, &err)
 
+	var rootResource string
+	var version string
+	var id string
+	parts := strings.Split(name, ":")
+	if len(parts) == 2 {
+		rootResource = parts[0]
+		version = parts[1]
+		id = rootResource + "-" + version
+	} else {
+		id = name
```

Comment:
In k8s,  will check `value, exists := labels["tag"]`. It means spec should contains rootResource and version before. So it will not by empty. And will add webhook validation for version and rootResource to ensure they both exists.

Reply:
Do we want to block object creation with label tag if rootResource and version are not specified? 
That's confusing.